File: app/src/main/java/com/hedgehog/gdzietabiedra/adapters/ShopsAdapter.java
package com.hedgehog.gdzietabiedra.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Filter;
import android.widget.Filterable;

import com.hedgehog.gdzietabiedra.R;
import com.hedgehog.gdzietabiedra.anim.AnimationUtils;

import com.hedgehog.gdzietabiedra.pojo.Shops.Shop;
import com.hedgehog.gdzietabiedra.utils.Api;
import com.hedgehog.gdzietabiedra.utils.Biedra;
import com.hedgehog.gdzietabiedra.viewHolders.ShopView;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

/**
 * Created by Adam on 2015-06-18.
 */
public class ShopsAdapter extends RecyclerView.Adapter<ShopView>{

    private List<Shop> mItems;
    private int mPreviousPosition;

    private final List<Shop> filteredItems;

    public ShopsAdapter(List<Shop> data) {
        mItems = data;
        this.filteredItems = new ArrayList<>();
    }

    @Override
    public ShopView onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(Biedra.getAppContext()).inflate(R.layout.shop_item, parent, false);
        return new ShopView(view);
    }

    @Override
    public void onBindViewHolder(ShopView holder, int position) {
        Shop item = mItems.get(position);
        holder.bindItem(item);
        if (position > mPreviousPosition) {
            AnimationUtils.animateFadeIn(holder, true);
        } else {
            AnimationUtils.animateFadeIn(holder, false);
        }
        mPreviousPosition = position;
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }
}

File: app/src/main/java/com/hedgehog/gdzietabiedra/adapters/ShopsAdapter.kt
package com.hedgehog.gdzietabiedra.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.hedgehog.gdzietabiedra.R
import com.hedgehog.gdzietabiedra.pojo.Shops.Shop
import com.hedgehog.gdzietabiedra.viewHolders.ShopView

/**
 * Created by Adam on 2015-06-18.
 */
class ShopsAdapter(private val mItems: List<Shop>) : RecyclerView.Adapter<ShopView>() {
    private var mPreviousPosition: Int = 0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ShopView {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.shop_item, parent, false)
        return ShopView(view)
    }

    override fun onBindViewHolder(holder: ShopView, position: Int) {
        val item = mItems[position]
        holder.bindItem(item)
        mPreviousPosition = position
    }

    override fun getItemCount(): Int {
        return mItems.size
    }
}
